platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/BasePlatformUnitTests.java

```diff
@@ -0,0 +1,115 @@
+        return createPlatformContext(null, null);
+    }
+
+    /**
+     * Creates a platform context for the tests with the given builder modifications. Modifications are applied in the
+     * following order:
+     * <ol>
+     *     <li>The config modifications are applied</li>
+     *     <li>The temp directory is added to the config for marker files</li>
+     *     <li>The platform context builder modifications are applied</li>
+     * </ol>
+     * Any configuration set by the platform context builder modifying method overrides the configuration created by by the config modifier. Best practice is to set configuration through the config modifier and all other platform context variables through the platform contect modifier.
+     *
+     * @param platformContextModifier the function to modify the platform context builder
+     * @param configModifier          the function to modify the test config builder
+     * @return the platform context
+     */
+    protected PlatformContext createPlatformContext(
+            @Nullable final Function<TestPlatformContextBuilder, TestPlatformContextBuilder> platformContextModifier,
+            @Nullable final Function<TestConfigBuilder, TestConfigBuilder> configModifier) {
+        final TestPlatformContextBuilder platformContextBuilder = TestPlatformContextBuilder.create();
+        final TestConfigBuilder configBuilder = new TestConfigBuilder();
+        if (configModifier != null) {
+            configModifier.apply(configBuilder);
+        }
+        // add temp directory to config for marker files.
+        configBuilder.withValue(
+                PathsConfig_.MARKER_FILES_DIR,
+                tempDir.resolve(TEST_MARKER_FILE_DIRECTORY).toString());
+        // add configuration to platform builder.
+        platformContextBuilder.withConfiguration(configBuilder.getOrCreateConfig());
+        if (platformContextModifier != null) {
+            // apply any other modifications to the platform builder.
+            platformContextModifier.apply(platformContextBuilder);
+        }
+        return platformContextBuilder.build();
+    }
+
+    /**
+     * Gets the temporary directory provided by JUnit
+     *
+     * @return the temporary directory
+     */
+    Path getTempDir() {
+        return tempDir;
+    }
+
+    /**
+     * Gets the marker file directory
+     *
+     * @return the marker file directory
+     */
+    protected Path getMarkerFileDirectory() {
+        return tempDir.resolve(TEST_MARKER_FILE_DIRECTORY);
+    }
+
+    /**
+     * checks for the presence of the marker file.
+     *
+     * @param markerFileName the name of the marker file
+     * @param exists         true if the marker file should exist, false otherwise
+     */
+    protected void assertMarkerFile(@NonNull final String markerFileName, final boolean exists) {
+        assertEquals(
+                exists,
+                getMarkerFileDirectory().resolve(markerFileName).toFile().exists());
+    }
```

Comment:
If this class is intended to be a class that all of our unit tests inherit from, I'd suggest we refrain from adding methods specific to individual features. 99.99% of tests won't need to interact with marker files.

Reply:
This PR definitely introduces tech debt that will have to be cleaned up.  Whether the method is here or in a utility file, I would expect that all unit tests that instantiate the event pipeline will want to turn marker file writing on and check that no marker files were written as the default behavior.